File: cses/leetcode/python/896.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        n = len(nums)
        inc, dec = False, False
        if n <= 2:
            return True
        for i in range(0,n-1):
            if nums[i]<nums[i+1]:
                inc = True
            elif nums[i]>nums[i-1]:
                dec = True
        return (inc or dec) and not (inc and dec) #we want to check whether inc is true or dec is true, but not both are true


arr = [1,2,2,3]
A = Solution()
A.isMonotonic(arr)

File: cses/leetcode/cpp/896.cpp

#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    bool isMonotonic(vector<int> &nums)
    {
        int n = nums.size();
        bool inc = false, dec = false;
        if (n <= 2)
            return true;
        for (int i = 0; i < n - 1; i++)
        {
            if (nums[i] < nums[i + 1])
            {
                inc = true;
            }
            else if (nums[i] > nums[i + 1])
            {
                dec = true;
            }
        }
        return inc ^ dec;
    }
};
int main()
{
    vector<int> nums = {6, 5, 4, 4};
    Solution A;
    A.isMonotonic(nums);
}